### PR TITLE
Skeletonization: Replace internally Polyhedron by HDS

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Replace the internal representation

## Release Management

* Affected package(s): Surface_mesh_skeleton

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <CGAL/extract_mean_curvature_flow_skeleton.h>
 #include <CGAL/iterator.h>
 #include <CGAL/Polygon_mesh_processing/connected_components.h>
+#include <CGAL/Polyhedron_items_with_id_3.h>
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp:31]: Do you really need that?

Comment (afabri) [reply Polyhedron/demo/Polyhedron/Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp:31]: I guess yes, but note that it is in the HDS package.

**Comment by sloriot** [comment]
@afabri Could this PR be responsible for that [runtime error](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.5-Ic-88/HalfedgeDS/TestReport_lrineau_Fedora-32-Release.gz)?

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-90

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp (hunk 1 above)
Comment (sloriot) [reply Polyhedron/demo/Polyhedron/Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp:31]: My point is that the demo uses surface mesh so why would the demo need that

Comment (afabri) [reply Polyhedron/demo/Polyhedron/Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp]: It was needed to provide the generic version for [this specialization](https://github.com/CGAL/cgal/pull/6547/files#diff-1e1f7269cc6361055ad7a3c053fe02225d77c9506833fceab57ab9fdf211f81bL43)  which I just removed as it is a no-op.

Comment (afabri) [reply Polyhedron/demo/Polyhedron/Plugins/PMP/Mean_curvature_flow_skeleton_plugin.cpp]: Cleaned with [eea9cf7](https://github.com/CGAL/cgal/pull/6547/commits/eea9cf7ac4e05ad6f49d94c0f2a7ca1864ffc74d)

**Comment by afabri** [comment]
> @afabri Could this PR be responsible for that [runtime error](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.5-Ic-88/HalfedgeDS/TestReport_lrineau_Fedora-32-Release.gz)?

It is not.   Fixed by #6604

**Comment by afabri** [comment]
Sh, maybe that was to first sort out the problem with the run time error.

**Comment by sloriot** [comment]
I added not yet approved because I want to have a look at why properties are duplicated and maybe it could be factorized with Polyhedron (like we did for triangulations and openmesh).

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-15

File: HalfedgeDS/package_info/HalfedgeDS/dependencies [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Algebraic_foundations
 BGL
 Circulator
+Distance_3
```
[the remark this answers is not in the document]
Comment (sloriot) [reply HalfedgeDS/package_info/HalfedgeDS/dependencies:4]: #include <CGAL/Distance_3/Point_3_Point_3.h> in properties.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-22

**Pull request merged**